app/src/main/java/de/danoeh/antennapod/fragment/SearchFragment.java

```diff
@@ -259,7 +259,15 @@ private void search() {
                     progressBar.setVisibility(View.GONE);
                     this.results = results.first;
                     adapter.updateItems(results.first);
-                    adapterFeeds.updateData(results.second);
+                    if(getArguments().getLong(ARG_FEED,0)==0)
+                    {
+                        adapterFeeds.updateData(results.second);
+                    }
+                    else
+                    {
+
+                        adapterFeeds.updateData(null);
```

Comment:
This line crashes because the list must not be `null`. The crash is only visible in the logcat output because it is a background task. It would be more clean to pass `Collections.emptyList()`.

Reply:
Understood, working on changing the code. Will see what the exact error is so I get a better understanding as to why it's wrong